cds_dojson/marc21/fields/books/base.py

```diff
@@ -615,14 +618,28 @@ def preprint_date(self, key, value):
 @model.over('number_of_pages', '^300__')
 def number_of_pages(self, key, value):
     """Translates number_of_pages fields."""
-    pages = clean_val('a', value, str)
-    if not pages:
+    val = clean_val('a', value, str)
+    if is_excluded(val):
         raise IgnoreKey('number_of_pages')
-    pages = re.search(r'(^[0-9]+) *p', pages)
-    if pages:
-        pages = int(pages.group(1))
-        return pages
-    raise UnexpectedValue(subfield='a')
+
+    if extract_physical_description(val):
+        self['physical_description'] = physical_description(self, key, value)
+
+    page_number = extract_page_number(val)
+    if page_number:
+        return page_number
+    raise IgnoreKey('number_of_pages')
```

Comment:
I don't see anything reported http://books-migrator-dev.web.cern.ch/results, @kprzerwa might have a better idea. My impression is that it gets silently omitted.

Reply:
I saw in the dojson code that `IgnoreKey` was designed and previously used to silently discard the field if none of the values found in MARC field should be migrated (for example if it does not fit the regexp). When this exception is raised, there is a handler for it which removes the field `number_of_pages` from the json

Reply:
@topless after discussion with @ntarocco we think that the rule should fail in the end (if it does not find expected values) because otherwise we might lose data